src/zigCompilerProvider.ts

```
@@ -1,101 +1,230 @@
+
+    childProcess.once("close", () => {
+      this.doASTGenErrorCheck.cancel();
+      this.astDiagnostics.delete(textDocument.uri);
+
+      if (stderr.length == 0) return;
+      var diagnostics: { [id: string]: vscode.Diagnostic[] } = {};
+      let regex = /(\S.*):(\d*):(\d*): ([^:]*): (.*)/g;
+
+      for (let match = regex.exec(stderr); match; match = regex.exec(stderr)) {
+        let path = textDocument.uri.fsPath;
+
+        let line = parseInt(match[2]) - 1;
+        let column = parseInt(match[3]) - 1;
+        let type = match[4];
+        let message = match[5];
+
+        let severity =
+          type.trim().toLowerCase() === "error"
+            ? vscode.DiagnosticSeverity.Error
+            : vscode.DiagnosticSeverity.Information;
+        let range = new vscode.Range(line, column, line, Infinity);
+
+        if (diagnostics[path] == null) diagnostics[path] = [];
+        diagnostics[path].push(new vscode.Diagnostic(range, message, severity));
+      }
+
+      for (let path in diagnostics) {
+        let diagnostic = diagnostics[path];
+        this.astDiagnostics.set(textDocument.uri, diagnostic);
+      }
+    });
+  }
+
+  private _doCompile(textDocument: vscode.TextDocument) {
+    let config = vscode.workspace.getConfiguration("zig");
+
+    let buildOption = config.get<string>("buildOption");
+    let processArg: string[] = [buildOption];
+    let workspaceFolder = vscode.workspace.getWorkspaceFolder(textDocument.uri);
+    if (!workspaceFolder && vscode.workspace.workspaceFolders.length) {
+      workspaceFolder = vscode.workspace.workspaceFolders[0];
+    }
+    const cwd = workspaceFolder.uri.fsPath;
+
+    switch (buildOption) {
+      case "build":
+        let buildFilePath = config.get<string>("buildFilePath");
+        processArg.push("--build-file");
+        try {
+          processArg.push(
+            path.resolve(buildFilePath.replace("${workspaceFolder}", cwd))
+          );
+        } catch {}
```

Not a big fan of empty catches but there were a few already so I guess it's okay?

---

I should add a comment explaining

`path.resolve` throws when the file is not found. The extension shouldn't crash in that case. But, a warning printed each time will make the editor feel slow, especially for something that is run often. 

Maybe it should print a warning just once.